### ISSUE TITLE
Strong paramter white list is not set correctly if a model is within a module

### ISSUE THREAD

**Issue by MPursche** [activeadmin/activeadmin#3694]
Hello!

I found a bug in the activeadmin DSL for resources. I am working on two applications that share some common models. I have put these shared models in a rails engine called SharedApplicationModel. A model definition could look something like this:
```ruby
module SharedApplicationModel
  class ModelSource < ActiveRecord::Base
    has_many :models
  end
end
```

I use the following code to integrate the model into active admin:
```ruby
ActiveAdmin.register SharedApplicationModel::ModelSource do
  menu :parent => "Model Sources", :label => "3D Models", :priority => 3
  permit_params :name, :description, :path, :preprocessed, :supported_styles
end
```

All views and controllers are generated accordingly, but if I want to create or update a ```ModelSource``` with the activeadmin website only an empty model is created. The problem is that permit_params sets the white list for ```params[:model_source]```. The automatically created form uses ```params[:model_source]``` as well. However, the default controller tries to read the parameters from ```params[:shared_application_model_model_source]```.
I was able to fix this issue for the forms by creating them with:
```ruby
form :as => 'shared_application_model_model_source' do |f|
 ...
end
```
But I found no way to correct the name for the strong parameter white list. As a workaround I implemented the create and update methods myself but this is not really DRY as I don't really do anything different from the default controller.

Greetings

Marcel

**Comment by risinglf** [comment]
+1

**Comment by zhirayrized** [comment]
+1

**Comment by azach** [comment]
+1

**Comment by seanlinsley** [comment]
This has been fixed since 8872d84ea906faed904b5baf129fd559cc3b650c.
